src/components/gltf-model-plus.js

```diff
@@ -11,6 +11,8 @@ import { RGBELoader } from "three/examples/jsm/loaders/RGBELoader";
 import { BasisTextureLoader } from "three/examples/jsm/loaders/BasisTextureLoader";
 import { DRACOLoader } from "three/examples/jsm/loaders/DRACOLoader";
 import { GLTFLoader } from "three/examples/jsm/loaders/GLTFLoader";
+import GLBRangeRequests from "three-gltf-extensions/loaders/GLB_range_requests/GLB_range_requests";
```

Comment:
https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L32
We should probably fetch the whole binary header here since we are going to need it anyway. The json chunk is also required to be first so we might as well also get its size/type too. So we can request bytes 0-20 in 1 request and then return the header, and first chunk length. This avoids 2 extra requests from the way its currently implemetned.

We should also check the byte length returned. If its larger than expected the server probably ignored the range header. In that case it would be ideal if we could just pass the buffer off to the normal loading flow so that we didn't need to download the whole file again.

https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L104
If we do the initial request to grab the size of the first chunk too this code gets simpler and no longer needs to be a loop. We just `loadPartially` the first chunk. The vinChunkOffset will just be the first chunk length + 12. If its the same as the header.length we know there is no binary chunk.

https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L121
Hmm not sure we should reject in this case. Its valid to have a glb with no bin, and we would not want to use a different loader just for that case (which we wouldn't know ahead of time).

https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L124
The way we are using GLBRangRequests right now this would mean we cant load anything with EXT_meshopt_compression at all anymore

https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L67
Dynamically registering this extension seems odd.. I guess there is no way for GLTFLoader plugins to override the initial load so we had to use this static load function instead?

Reply:
General notes:
Slightly concerned about increasing the number of requests in the (very common) case that the glb does not have any LODs that need progressive loading. I think my suggestions knock it down to 2 requests instead of 4 but that's still an extra request.

Reply:
> We should also check the byte length returned. If its larger than expected the server probably ignored the range header. 

`FileLoader` invokes `onError` if the server doesn't support HTTP range requests. And if the plugin catches the error it loads a glTF file in the conditional way (without range requests).

https://github.com/mrdoob/three.js/pull/24580

> so that we didn't need to download the whole file again.

I assume browser's disk cache works

Reply:
> https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L124
The way we are using GLBRangRequests right now this would mean we cant load anything with EXT_meshopt_compression at all anymore

Ah, I was going to

1. rejects if `EXT_meshopt_compression` is found
2. and then load the asset with regular `gltfLoader.load()`

but I have wrongly implemented. I will update the implementation.

So that the combination of progressive loading + `EXT_meshopt_compression` is impossible but `EXT_meshopt_compression` can still be used. progressive loading + `EXT_meshopt_compression` is a future work for me.

Reply:
> https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L121
Hmm not sure we should reject in this case. Its valid to have a glb with no bin, and we would not want to use a different loader just for that case (which we wouldn't know ahead of time).

Makes sense to me. Updated.

Reply:
> https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L32
We should probably fetch the whole binary header here since we are going to need it anyway. The json chunk is also required to be first so we might as well also get its size/type too. So we can request bytes 0-20 in 1 request and then return the header, and first chunk length. This avoids 2 extra requests from the way its currently implemetned.

> https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L104
If we do the initial request to grab the size of the first chunk too this code gets simpler and no longer needs to be a loop. We just loadPartially the first chunk. The vinChunkOffset will just be the first chunk length + 12. If its the same as the header.length we know there is no binary chunk.

Sounds good to me. Updated.

Reply:
> https://github.com/takahirox/three-gltf-extensions/blob/62288914fbafb2500e500b0209b33d4f3a96cfac/loaders/GLB_range_requests/GLB_range_requests.js#L67
Dynamically registering this extension seems odd.. I guess there is no way for GLTFLoader plugins to override the initial load so we had to use this static load function instead?

Your guess is right. I couldn't come up with any other ideas.

Reply:
Added a patch for avoiding duplicated full range content request in case of server doesn't support range requests.

https://github.com/takahirox/three-gltf-extensions/pull/90